### DISCUSSION TITLE
Unable to fetch JSON file

### DISCUSSION THREAD

**Discussion by jkohlin** [rdilweb/docusaurus-plugin-remote-content#43] [Q&A] [answered]
I'm running Docusaurus in a Docker container and I can't get this plugin to work. I get this error in the server console:
```
[ERROR] Error: connect ECONNREFUSED 127.0.0.1:443 at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1187:16)
``` 
I tried to add the proxy-settings to the requestConfig arg but still doesn't work. 
If I start Docusaurus with this plugin out commented, and uncomment it after the server starts, I at least won't get the error above, but instead I get a 404.

```
requestConfig: {
  proxy: {
    protocol: 'http',
    host: '0.0.0.0',
    port: 3000,
},
    responseType: 'json',
    headers: {
      'Access-Control-Allow-Origin': '*'
    }
  }
```
You can try it out here: 
https://codesandbox.io/s/brave-worker-n3608u?file=/docusaurus.config.js

**Comment by RDIL** [comment, marked as answer]
<img width="885" alt="Screen Shot 2022-10-07 at 13 54 01" src="https://user-images.githubusercontent.com/34555510/194619852-8af57c2a-7393-472f-9e05-13712f0a5d78.png">

Looks like an issue with your asset server.

**Reply by jkohlin** [reply]
Sorry, just pasted in my dev code, so this was my dev server. 
When applying a public url it worked.

BUT, the json is saved to file like this
```json
"{\n  \"guld.css\": \"/guld.c837d7d5c7b640834d80.css\",\n  \"guld.js\": \"/guld.6a772a8422ec04d5cfa7.js\"}"
```
I don't want the enclosing `"` quotes in the file. and I don't want the escaped quotes in the json code.
Is that possible?

**Reply by RDIL** [reply]
Try setting the `responseType` to `text` - that should fix it.

**Reply by jkohlin** [reply]
Sorry, nope. This is my config:
```javascript
[
  "docusaurus-plugin-remote-content",
  {
    name: "manifest", // used by CLI, must be path safe
    sourceBaseUrl: "http://ms_dev_assets/", // the base url for the markdown (gets prepended to all of the documents when fetching)
    outDir: "src/utils", // the base directory to output to.
    documents: ["manifest.json"], // the file names to download
    requestConfig: {
      transformResponse: (data) => JSON.stringify(data, null, 4),
      responseType: 'text',
      headers: {
        'Content-Type': 'application/json'
      }
    },
  }
],

```

**Reply by RDIL** [reply]
Honestly, it looks like this is just an issue with your setup, unless you can provide a full reproduction (including a working public asset server and a Docusaurus config), there isn't anything I can do.

**Reply by jkohlin** [reply]
Well, i did modify the sandbox so you can see it there. 

https://codesandbox.io/s/brave-worker-n3608u?from-embed=&file=/docusaurus.config.js

The manifest.json file that your plugin creates is in the src folder.

**Reply by RDIL** [reply]
Found an answer - remove the `transformResponse`, and switch the `responseType` to `arraybuffer` - actually confirmed that it does work this time.

**Reply by jkohlin** [reply]
Thank you so much! 
That was it!
